File: server/router/hospitalManage/prescriptionQualification.go
package hospitalManage

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type PrescriptionQualification struct{}

// 职称管理
func (j *JobTitle) InitPPRouter(Router *gin.RouterGroup, RouterPub *gin.RouterGroup) {
	ppRouter := Router.Group("prescription").Use(middleware.OperationRecord())
	ppApi := v1.ApiGroupApp.HospitalManageApiGroup.PrescriptionQualification
	{
		ppRouter.GET("getList", ppApi.GetListPQ)
		ppRouter.POST("create", ppApi.CreatePQ)
		ppRouter.POST("update", ppApi.UpdatePQ)
		ppRouter.DELETE("delete", ppApi.DeletePQ)
		ppRouter.POST("addPersonPQItem", ppApi.AddPersonPQItem)
		ppRouter.GET("getPersonPQList", ppApi.GetPersonPQList)
		ppRouter.DELETE("deletePersonPQ", ppApi.DeletePersonPQ)
	}
}

File: web/src/api/prescriptionQualification.js
import service from "@/utils/request";

export const getListPQ = (params) => {
  return service({
    url: "/prescription/getList",
    method: "get",
    params,
  });
};
export const createPQ = (data) => {
  return service({
    url: "/prescription/create",
    method: "post",
    data,
  });
};
export const updatePQ = (data) => {
  return service({
    url: "/prescription/update",
    method: "post",
    data,
  });
};
export const deletePQ = (params) => {
  return service({
    url: "/prescription/delete",
    method: "delete",
    params,
  });
};
export const addPersonPQ = (data) => {
  return service({
    url: "/prescription/addPersonPQItem",
    method: "post",
    data,
  });
};
export const getPersonPQList = (params) => {
  return service({
    url: "/prescription/getPersonPQList",
    method: "get",
    params,
  });
};
export const deletePersonPQ = (params) => {
  return service({
    url: "/prescription/deletePersonPQ",
    method: "delete",
    params,
  });
};
